utils/prompt_generation.py

```diff
@@ -0,0 +1,424 @@
+    # re-apply template to truncated prompts
+    # finally, template the prompts and return prompt text with templating
+    templated_trunc_prompts = template_prompt(
+        trunc_prompts,
+        template=template,
+        tokenizer=tokenizer,
+        tokenize=False,
+        tokenizer_model=tokenizer_model,
+    )
+    return templated_trunc_prompts
+
+
+def process_prompts(prompts, max_length, template, tokenizer_model):
+    tokenizer = AutoTokenizer.from_pretrained(tokenizer_model)
+    if template:
+        # manually apply template to prompts
+        logging.info(f"Applying template to the generated prompts: {template}")
+        truncated_prompts = truncate_template_prompt(
+            prompts, tokenizer, template, max_length, tokenizer_model
+        )
+    else:
+        logging.info("No template applied to generated prompts.")
+        truncated_prompts = [
+            tokenizer.decode(
+                tokenize_encode(
+                    p,
+                    tokenizer=tokenizer,
+                    max_length=max_length,
+                    tokenizer_model=tokenizer_model,
+                )
+            )
+            for p in prompts
+        ]
+
+    processed_prompts = truncated_prompts
+
+    processed_lengths = [
+        len(
+            tokenize_encode(
+                p, tokenizer=tokenizer, max_length=None, tokenizer_model=tokenizer_model
+            )
+        )
+        for p in processed_prompts
+    ]
+    return processed_prompts, processed_lengths
+
+
+# Main function to handle prompt generation and templating
+def generate_prompts(args):
+    logging.info(f"generate_prompts args={args}")
+    # vLLM appears to add extra token on receipt of prompt
+    # TODO: verify if this is bos token or something else
+    args.max_prompt_length = args.max_prompt_length - 1
+    if args.input_seq_len == -1:
+        args.input_seq_len = args.max_prompt_length
+    else:
+        args.input_seq_len = args.input_seq_len - 1
+
+    if args.dataset.lower() == "random":
+        # default case
+        logger.info("Generating random prompts...")
+        # -1 is for the extra token added by vLLM
+        assert args.input_seq_len > -1, "input_seq_len must be set for random prompts."
+        assert args.max_prompt_length > -1, "max_length must be set for random prompts."
+        prompts = generate_random_prompts(
+            args.num_prompts,
+            args.max_prompt_length,
+            args.input_seq_len,
+            args.distribution,
+            args.tokenizer_model,
+        )
+    elif args.dataset is not None:
+        assert (
+            args.max_prompt_length > -1
+        ), "max_length must be set for datasets prompts."
+        logger.info(f"Generating prompts from the '{args.dataset}' dataset...")
+        if args.dataset == "alpaca_eval":
+            prompts = load_alpaca_eval_dataset_samples(args.num_prompts)
+        elif args.task is not None:
```

Comment:
What should the user set `--dataset` to if they want to generate from a `--task`?

If they set `--dataset random` they will generate the random set and if they set `--dataset alpaca_eval` they will get the AlpacaEval dataset irrespective of what `--task` is set to. But, if they want a `--tast` set, what should `--dataset` be?

Reply:
I added a commit to remove the --task support, I think it's overly complicated to have LM eval prompt templating in this CLI. I didnt find it very useful, can add it back if there's a real use case for testing.